contracts/SimpleAuthorization.sol

```diff
@@ -6,25 +6,35 @@ import "./TokenValidator.sol";
 // Reference Validator
 
 contract SimpleAuthorization is TokenValidator, Owned {
-  mapping(address => bool) private authorizations;
+    mapping(address => bool) private auths;
 
-  function SimpleAuthorization() public {
-  }
+    function SimpleAuthorization() public Owned {}
 
-  function check(address /* token */, address _address) public /* view */ returns (uint8 resultCode) {
-      return authorizations[_address] ? 1 : 0;
-  }
+    function check(
+        address /* token */,
+        address _address
+    ) public /* view */ returns (byte resultCode) {
+        if (auths[_address]) {
+            return hex"11";
+        } else {
+            return hex"10";
+        }
```

Comment:
Not totally sure _why_ it didn't like this as a ternary operator. I've run into this a few times.

Reply:
I tried the following variations:

```solidity
(auths[_address]) ? return hex"11" :  return hex"10";
(auths[_address]) ? (return hex"11") :  (return hex"10");
return (auths[_address]) ? hex"11" :  hex"10";
return ((auths[_address]) ? hex"11" :  hex"10");
return ((auths[_address]) ? (hex"11") :  (hex"10"));
```

None of these compiled. It complained that I wasn't returning a `byte`. Switch to `if/else` syntax works 🤷🏻‍♀️

Reply:
weird - maybe a solidity shortcoming.  or maybe disdain for the elvis operator.  reading this: http://solidity.readthedocs.io/en/v0.4.21/control-structures.html#output-parameters

i suspect that `result = auths[_address] ? 1 : 0;` might actually work.  or even, simply `auths[_address] ? 1 : 0;` - ie, skip the return as it is the last line

Reply:
i'm not really a fan of the implict return.  i'm know i'm not a rubyist.

Reply:
> i suspect that result = auths[_address] ? 1 : 0; might actually work

I've had this work with assignment, yeah

> simply auths[_address] ? 1 : 0; - ie, skip the return as it is the last line

I'm pretty sure that you need an explicit `return`. I see it in all the one-liners, for instance.

Reply:
btw, none of this is a big deal, just trying to understand the quirks of solidity.